src/lib/settings/schema/SchemaEntry.js

```diff
@@ -234,19 +241,34 @@ class SchemaEntry {
 	 * @private
 	 */
 	_generateDefault() {
-		if (this.array) return [];
+		// TODO(kyranet): How do we generate the defaults?
+		// if (this.array) return [];
 		if (this.type === 'boolean') return false;
 		return null;
 	}
 
+	/**
+	 * Infers the correct group given the default
+	 * @since 0.5.0
+	 * @param {*} defaultValue The default value to infer
+	 * @returns {Function}
+	 * @private
+	 */
+	_inferGroupFromDefault(defaultValue) {
+		if (Array.isArray(defaultValue)) return SettingsArray;
+		if (defaultValue instanceof Map) return SettingsMap;
+		// if (defaultValue instanceof Set) return SettingsSet;
+		return null;
+	}
+
 	/**
 	 * Get a JSON object containing data from this SchemaEntry
 	 * @since 0.5.0
 	 * @returns {Object}
 	 */
 	toJSON() {
 		return {
-			array: this.array,
+			group: this.group ? this.group.name : null,
```

Comment:
We need to discuss this more... need to get a good balance between leaving the properties intact but also not filling out the entire property accessors